src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -481,6 +461,42 @@ class CosmasGoogleCloudService(private val freeBucketName: String,
         responseObserver.onCompleted()
     }
 
+    fun restoreFileFromStorage(fileId: String, projectInfo: ProjectInfo,
+                               project: ConcurrentMap<String, CosmasProto.FileVersion>): FileVersion {
+        synchronized(project) {
+            // Getting last version from storage or default instance if it doesn't exist
+            val latestVersionBlob: Blob? = this.storage.get(getBlobId(fileId, projectInfo))
+            val latestVersion = if (latestVersionBlob != null) {
```

Comment:
This combination of `if` with `return` looks uncomfortable. I suggest appending null-case to the previous line with Elvis-operator:

```
            val latestVersionBlob: Blob = this.storage.get(getBlobId(fileId, projectInfo)) ?: return FileVersion.getDefaultInstance() 
            val latestVersion = CosmasProto.FileVersion.parseFrom(latestVersionBlob.getContent())
```

Reply:
ok